### PR TITLE
FIX TextUnrestricted recursively applies to JSON objects and array

### PR DESCRIPTION
**Pull request by Anjali-NEC**
Fix issue #3867

### CODE REVIEW

**Comment by Anjali-NEC** [comment]
Added test case in commit https://github.com/telefonicaid/fiware-orion/pull/3919/commits/20874a6fe8ac1bae10f4235ff87cdada42ce4da4

File: test/functionalTest/cases/3867_textUnrestricted_should_be_recursively_applied_to_JSON_objects_and_array/textUnrestricted_should_be_recursively_applied_to_JSON_objects_and_array.test [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+TextUnrestricted should be recursively applied to JSON objects and array
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. POST /v2/entities with attribute type TextUnrestricted
```
Comment (fgalan) [inline test/functionalTest/cases/3867_textUnrestricted_should_be_recursively_applied_to_JSON_objects_and_array/textUnrestricted_should_be_recursively_applied_to_JSON_objects_and_array.test:33]: I'd suggest to add a second step to retrieve the entity, i.e:

```suggestion
# 01. POST /v2/entities with attribute type TextUnrestricted
# 02. GET /v2/entities to check the TextUnrestricted fields are correctly rendered
```

Comment (Anjali-NEC) [reply test/functionalTest/cases/3867_textUnrestricted_should_be_recursively_applied_to_JSON_objects_and_array/textUnrestricted_should_be_recursively_applied_to_JSON_objects_and_array.test:33]: Fixed in https://github.com/telefonicaid/fiware-orion/pull/3919/commits/4e3c1658dea590ac07c9e7e2faa5c7a233d41afd

**Comment by fgalan** [comment]
Good work! :)

I think the only thing missing to finish the PR is to add to CHANGES_NEXT_RELEASE file a line about the change done. Something like that could be:

```
- Fix: TextUnrestricted recursively applies to JSON objects and array (#3867)
```

**Review by fgalan — APPROVED** [review]
LGTM. Thanks for your contribution!

**Pull request merged**